bundles/org.openhab.binding.remoteopenhab/src/main/java/org/openhab/binding/remoteopenhab/internal/handler/RemoteopenhabBridgeHandler.java

```diff
@@ -0,0 +1,554 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Undefined REST URL setting in the thing configuration");
+            return;
+        }
+        URL url;
+        try {
+            url = new URL(config.restUrl.trim());
+        } catch (MalformedURLException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Invalid REST URL setting in the thing configuration");
+            return;
+        }
+        if (!"http".equals(url.getProtocol())) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Use HTTP protocol for the REST URL setting in the thing configuration");
+            return;
+        }
+        List<String> localIpAddresses = NetUtil.getAllInterfaceAddresses().stream()
+                .filter(a -> !a.getAddress().isLinkLocalAddress())
+                .map(a -> a.getAddress().getHostAddress().split("%")[0]).collect(Collectors.toList());
+        if (localIpAddresses.contains(url.getHost().replaceAll("\\[|\\]", ""))) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Do not link the local server with the REST URL setting in the thing configuration");
+            return;
+        }
+
+        String urlStr = url.toString();
+        if (urlStr.endsWith("/")) {
+            urlStr = urlStr.substring(0, urlStr.length() - 1);
+        }
+        logger.debug("REST URL = {}", urlStr);
+
+        RemoteopenhabRestClient client = new RemoteopenhabRestClient(clientBuilder, eventSourceFactory, jsonParser,
+                config.token, urlStr);
+        restClient = client;
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        startCheckConnectionJob(client);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing remote openHAB handler for bridge {}", getThing().getUID());
+        stopStreamingUpdates();
+        stopCheckConnectionJob();
+        this.restClient = null;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (getThing().getStatus() != ThingStatus.ONLINE) {
+            return;
+        }
+        RemoteopenhabRestClient client = restClient;
+        if (client == null) {
+            return;
+        }
+
+        try {
+            if (command instanceof RefreshType) {
+                String state = client.getRemoteItemState(channelUID.getId());
+                updateChannelState(channelUID.getId(), null, state);
+            } else if (isLinked(channelUID)) {
+                client.sendCommandToRemoteItem(channelUID.getId(), command);
+                String commandStr = command.toFullString();
+                logger.debug("Sending command {} to remote item {} succeeded",
+                        commandStr.length() < 50 ? commandStr : commandStr.substring(0, 50) + "...",
```

Comment:
Can you turn the [magic number](https://en.wikipedia.org/wiki/Magic_number_(programming)) `50` into a constant with a descriptive name? It's also used below on line 541.

Reply:
Done